### PR TITLE
Updating Install

### LINKED ISSUE #710: INSTALL.txt
Some changes INSTALL.txt. 
Just let me know if you have any suggestions before I submit a pull request. (I am not a mac user).

`autoconf` should be present on your machine
  (if not we get path-spec problems)
Should be included in the list under 1.

We could convert INSTALL.txt into markdown.

OSX
opam can be install by
   brew install opam
and then
   brew update; brew upgrade objective-caml

In step 3, we should advise -j2, where 2 is the number of cores you have on your machine.

### PR DESCRIPTION
**Pull request by spitters**
The git tutorial. If it's a problem I'll send a new pull request.
This closes #710

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -13,61 +23,64 @@ minimal fuss, you should try your luck by following these instructions:
...
        make
-
-   It may take a while to compile the custom Coq.
-
+      ```
+   It may take a while to compile the custom Coq. To speed this up, use `make -jn`, where n is the number of cores you have on your machine.
+   On linux this can be found with `nproc` or `lscpu`. 
+   On OSX Apple menu -> About this Mac -> System Report, then look for "number of cores"
 
    If you are using Debian/Ubuntu, and don't mind not having HoTT/coq
```
Comment (mikeshulman) [inline INSTALL.md]: Not related to this PR, but should this be "don't mind having"?

File: git.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+`git reset --hard origin/master `
+
+We have our own version of Coq in a git submodule. When it is updated upstream run:
+`etc/install_coq.sh `
+or by hand:
+`cd coq-HoTT && make clean `
+or, if that's not enough, 
+`cd coq-HoTT && git clean -xfd`
+Then
+`git submodule update && cd coq-HoTT && configure -local && make -j2 -k coqide coqlight`
```
Comment (mikeshulman) [inline git.md]: These instructions look a little confusing to me.  The automatic script `etc/install_coq.sh` does all of the subsequently listed commands (clean (if necessary) then configure then make), so I think it could be clearer that "by hand" scopes over the entire rest of the paragraph.  Also, after someone has done `cd coq-HoTT && make clean` I think they will need to `cd ..` before `git submodule update` (or else run the update before the clean), and on most unices they'll also need to say `./configure` (`.` isn't usually in the `PATH`).

**Comment by mikeshulman** [comment]
LGTM modulo my comment.  Thanks!

File: INSTALL.md [hunk 3]
```diff
@@ -13,61 +23,64 @@ minimal fuss, you should try your luck by following these instructions:
      which installs the dependencies automatically.
 
    * OSX: we recommend that you install the Opam package manager for Ocaml
-     (http://opam.ocamlpro.com/) and go from there.
-
+     (`http://opam.ocamlpro.com/`) and go from there.
```
Comment (JasonGross) [inline INSTALL.md]: URLs should not be between backticks; that makes them unclickable in markdown.

File: INSTALL.md [hunk 4]
```diff
@@ -13,61 +23,64 @@ minimal fuss, you should try your luck by following these instructions:
...
+   ```
+     brew install opam
```
Comment (JasonGross) [inline INSTALL.md]: These commands should probably be unindented, if they're between triple backticks.

File: INSTALL.md (hunk 1 above)
Comment (JasonGross) [inline INSTALL.md]: I agree.

File: git.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
+Elementary git usage. This is in addition to the instructions on how to set up git in INSTALL.md
+
+USAGE 1: add + commit
+To have git record your changes:
```
Comment (JasonGross) [inline git.md]: Markdown seems to treat single newlines as spaces.  The looks strange without punctuation:
> USAGE 1: add + commit To have git record your changes: `git commit -m "description of edit" <files-you-changed>`
>
> USAGE 2: pull To pull changes by others from GitHub: (Your local directory will still have your changes.) We call the central repo "upstream" `git add remote upstream https://github.com/HoTT/HoTT.git` FIRST commit your changes as above, then `git pull upstream master` (This is the central repo)

File: git.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+See [Contributing](https://github.com/HoTT/HoTT/blob/master/STYLE.md)
+To push your changes to GitHub:
+FIRST, commit your changes as above
+SECOND, pull changes from GitHub as above, then
+```
+make -j2
+git push origin master
+```
+(This is your own repo)
+Now go to your github page and click create pull requests.
```
Comment (JasonGross) [inline git.md]: Nit: "create pull request", not "create pull request**s**"

File: git.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+This will then be compiled (again) by Travis and reviewed by the developers.
+After github discussion, and possible updates of the pull request, 
+one of them will pull your request.
+
+*Only very minor* changes (spelling) can be pushed directly.
```
Comment (JasonGross) [inline git.md]: Do you want to note that this can only be done by the repo admins/owners?

File: git.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+`git push upstream master`
+
+Which files did I change since the last commit?
+`git status`
+
+What changes did I make in those files?
+`git diff`
+
+Who wrote in this file?
+`git praise`
```
Comment (JasonGross) [inline git.md]: I think `git blame` is more commonly uses, but, *shrugs*

File: git.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+We have our own version of Coq in a git submodule. When it is updated upstream run:
+`etc/install_coq.sh `
+or by hand:
+`cd coq-HoTT && make clean `
+or, if that's not enough, 
+`cd coq-HoTT && git clean -xfd`
+Then
+`git submodule update && cd coq-HoTT && configure -local && make -j2 -k coqide coqlight`
+
+For more, there are [plenty][1] of [tutorials][2] online
```
Comment (JasonGross) [inline git.md]: Nit: Missing a period at the end of the sentence.

File: git.md (hunk 5 above)
Comment (JasonGross) [inline git.md]: Note: You can see a preview here: https://github.com/spitters/HoTT/blob/install/git.md

File: INSTALL.md (hunk 1 above)
Comment (spitters) [inline INSTALL.md]: What would you prefer?

**Comment by spitters** [comment]
I think I made all the changes.

File: INSTALL.md (hunk 1 above)
Comment (mikeshulman) [inline INSTALL.md]: Might as well fix it here if you want, otherwise someone can do it in another PR.

File: INSTALL.md [hunk 10]
```diff
@@ -1,3 +1,13 @@
+# EXPERIMENTAL INSTALL INSTRUCTIONS
+We have experimental opam packages to install both the correct Coq version and the library[1].
+However, the latter cannot be used by people using git to contribute to the library.
+Moreover, opam support on windows is experimental.
+[1]: https://github.com/HoTT/HoTT/issues/694
+
+Moreover, we are compatible with [Coq8.5beta](https://coq.inria.fr/coq-85), so binary packages can be used.
```
Comment (JasonGross) [inline INSTALL.md]: Nit: Maybe add spaces, so `Coq 8.5 beta` rather than `Coq8.5beta`

**Comment by spitters** [comment]
Done.

File: INSTALL.md [hunk 11]
```diff
@@ -1,3 +1,13 @@
+# EXPERIMENTAL INSTALL INSTRUCTIONS
+We have experimental opam packages to install both the correct Coq version and the library[1].
+However, the latter cannot be used by people using git to contribute to the library.
+Moreover, opam support on windows is experimental.
+[1]: https://github.com/HoTT/HoTT/issues/694
+
+Moreover, we are compatible with [Coq8.5 beta](https://coq.inria.fr/coq-85), so binary packages can be used.
```
Comment (andrejbauer) [inline INSTALL.md]: Maybe avoid the sequence "However ... Moreover ... Moreover ... However"?

**Comment by spitters** [comment]
Done

**Comment by spitters** [comment]
@andrejbauer are you happy with this?

**Comment by andrejbauer** [comment]
I have some questions.

1. Why are you formatting displayed code with backquotes instead of indenting it by 4 characters?
2. `git.md` looks a bit ugly and actually not that helpful at all. Why do we even have it? There are a gazillion git tutorials out there.

**Comment by spitters** [comment]
Thanks. I did not know about the 4 space convention. I can change it if you want.

I used the git.md for the Coq group here at CMU. It might be useful for beginners, but I could remove it.

**Comment by andrejbauer** [comment]
I would like to hear another opinion about `git.md`. If it gets formatted a bit better and others think it's useful, we can keep it.

**Comment by mikeshulman** [comment]
I don't care one way or the other.

**Comment by spitters** [comment]
The four spaces convention seems to be specific to git's md. E.g. the
retext editor does not support it.
Do we still want to follow this convention? I'd be happy to make the change.

**Comment by andrejbauer** [comment]
Sure, we use it elsewhere and who's going to read the file with retext? It'll either be raw file or github web site.

**Comment by spitters** [comment]
Done. @JasonGross do you want git.md or not. We can't decide.

**Comment by JasonGross** [comment]
Regarding spaces vs quotes, I will note that triple quotes supports language-specific syntax highlighting on github, as in "```coq", which is perhaps a reason to prefer that, at least if you want syntax highlighting.

I have no strong feelings about `git.md`.  In most programming projects, I would be against having it, but if we're catering to beginners, it might be useful.  I echo the "why not just point to some popular tutorials on the web?" concern, but would not really object to having it added.

**Comment by spitters** [comment]
OK. I've removed git.md as there seems to be very little enthusiasm for it.

Otherwise, I think we are done, aren't we?

File: INSTALL.md [hunk 12]
```diff
@@ -13,10 +22,13 @@ minimal fuss, you should try your luck by following these instructions:
      which installs the dependencies automatically.
 
    * OSX: we recommend that you install the Opam package manager for Ocaml
-     (http://opam.ocamlpro.com/) and go from there.
+(http://opam.ocamlpro.com/) and go from there.
```
Comment (mikeshulman) [inline INSTALL.md]: What happened to the indentation here?

File: INSTALL.md [hunk 13]
```diff
@@ -102,21 +114,21 @@ minimal fuss, you should try your luck by following these instructions:
...
    sessions`.
 
-   	(let ((default-directory
-   	       (locate-dominating-file buffer-file-name ".dir-locals.el")))
-   	  (make-local-variable 'coq-prog-name)
-   	  (setq coq-prog-name (expand-file-name "../hoqtop")))
+    (let ((default-directory
+         (locate-dominating-file buffer-file-name ".dir-locals.el")))
+    (make-local-variable 'coq-prog-name)
+    (setq coq-prog-name (expand-file-name "../hoqtop")))
```
Comment (mikeshulman) [inline INSTALL.md]: The internal relative indentation of the elisp code got lost here.

**Comment by spitters** [comment]
@mikeshulman Thanks. Done.

**Comment by mikeshulman** [comment]
Ok, still LGTM.

File: INSTALL.md [hunk 14]
```diff
@@ -25,48 +37,48 @@ minimal fuss, you should try your luck by following these instructions:
      `dos2unix`.  The following commands, from the root of the HoTT
      library, should mostly take care of things:
 
-       dos2unix hoq*
-       /usr/bin/find . -name '*.sh' | xargs dos2unix
+    dos2unix hoq*
+    /usr/bin/find . -name '*.sh' | xargs dos2unix
```
Comment (JasonGross) [inline INSTALL.md]: These are not sufficiently indented to appear as code/on separate lines.  Add triple quotes or enough spaces to fix the display when you click "view".

File: INSTALL.md [hunk 15]
```diff
@@ -13,10 +22,13 @@ minimal fuss, you should try your luck by following these instructions:
      which installs the dependencies automatically.
 
    * OSX: we recommend that you install the Opam package manager for Ocaml
-     (http://opam.ocamlpro.com/) and go from there.
+(http://opam.ocamlpro.com/) and go from there.
+
+    brew install opam
+    brew update; brew upgrade objective-caml
```
Comment (JasonGross) [inline INSTALL.md]: These display all on a single line.

File: INSTALL.md [hunk 16]
```diff
@@ -25,48 +37,48 @@ minimal fuss, you should try your luck by following these instructions:
...
 
      If you wish to build CoqIDE/hoqide on Windows, we wish you good
      luck.
 
 2. Get the HoTT library (skip this step if you already have it):
 
-       git clone https://github.com/HoTT/HoTT.git
-       cd HoTT
+    git clone https://github.com/HoTT/HoTT.git
+    cd HoTT
```
Comment (JasonGross) [inline INSTALL.md]: These also display on the same line

**Comment by JasonGross** [comment]
The indentation still seems broken, as per https://github.com/spitters/HoTT/blob/install/INSTALL.md.

**Comment by spitters** [comment]
Unless anyone can give me a link to some same semantics for lists interacting with code for MD on git, I am done now.

**Comment by spitters** [comment]
Looks fine now. Found an online editor to test.
